src/main/kotlin/org/neo4j/graphql/handler/projection/ProjectionBase.kt

```diff
@@ -16,21 +16,30 @@ open class ProjectionBase {
     }
 
     fun orderBy(variable: String, args: MutableList<Argument>): String {
+        val values = getOrderByArgs(args)
+        if (values.isEmpty()) {
+            return ""
+        }
+        return " ORDER BY " + values.joinToString(", ", transform = { (property, direction) -> "$variable.$property $direction" })
+    }
+
+    private fun getOrderByArgs(args: MutableList<Argument>): List<Pair<String, Sort>> {
         val arg = args.find { it.name == ORDER_BY }
-        val values = arg?.value?.let { it ->
-            when (it) {
-                is ArrayValue -> it.values.map { it.toJavaValue().toString() }
-                is EnumValue -> listOf(it.name)
-                is StringValue -> listOf(it.value)
-                else -> null
+        return arg?.value
+            ?.let { it ->
+                when (it) {
+                    is ArrayValue -> it.values.map { it.toJavaValue().toString() }
```

Comment:
don't we need to check here as well Enum/StringValue?

Reply:
`it.toJavaValue()` does the check